### PR TITLE
Sort lists

### PR DESCRIPTION
**Pull request by steveri**
As promised...this small set of proposed changes improves the repeatability of mflowgen output, basically by sorting various random lists before attempting to print them to stdout...see file-compares for details.

### CODE REVIEW

File: mflowgen/utils/__init__.py
Comment (steveri) [inline mflowgen/utils/__init__.py:7]: This of course has nothing to do with the sorted lists...I just noticed that one of the methods had been left out of utils `__init__.py` and decided to make things consistent...

**Review by ctorng — CHANGES_REQUESTED** [review]

File: mflowgen/components/__init__.py
Comment (steveri) [reply mflowgen/components/__init__.py:3]: That makes sense, I will reset the file.

File: mflowgen/utils/__init__.py
Comment (steveri) [reply mflowgen/utils/__init__.py:7]: Fair enough.

File: mflowgen/components/graph.py
Comment (steveri) [reply mflowgen/components/graph.py:649]: yep, that's better

File: mflowgen/backends/makefile_syntax.py
Comment (steveri) [reply mflowgen/backends/makefile_syntax.py:61]: cool.

**Comment by steveri** [comment]
@ctorng  thanks very much for the feedback. I think I have addressed all your concerns; when you have a chance, please have another look and see if it's okay. Once I get your approval, I will finalize the merge.

File: mflowgen/components/graph.py
Comment (ctorng) [inline mflowgen/components/graph.py:127]: This should be edges_o, not edges_0.

**Comment by ctorng** [comment]
After those fixes, I think it looks great!

We should "Squash and merge" so there is one commit for this rather than 39 commits.

File: mflowgen/components/graph.py
Comment (steveri) [reply mflowgen/components/graph.py]: Well that's disturbing. I had no idea that I had changed code that was not being tested :(
So then after I fixed the typo, I wrote a test that did exercise the code, and didn't particularly like what I saw, so changed the sort to use `dst` as the key instead of `src,` as that order seemed visually more pleasing.
I don't think it matters in the end, and any order is better than none I guess; but anyway I will give you a couple days to respond to this change, if you like, before I do the final merge.

**Comment by steveri** [comment]
> After those fixes, I think it looks great!
> 
> We should "Squash and merge" so there is one commit for this rather than 39 commits.

Thanks Chris. Yes I will definitely squash when I merge, most of those commits had *nothing* to do with the final product...

**Review by ctorng — APPROVED** [review]

File: mflowgen/components/graph.py
Comment (ctorng) [reply mflowgen/components/graph.py]: Looks great, feel free to merge!

**Pull request merged**
